### PR TITLE
[154] Add Fastlane

### PR DESCRIPTION
**Pull request by llleyelll**
Resolves https://github.com/nimblehq/ios-templates/issues/154

## What happened

- Add `TestManager`
- Add clean up lane
- Add `Gemfile`
 
## Insight

- Update test function to test only specific targets (tests or UI tests).
- Code signing has been implemented in #157 
 
## Proof Of Work

Screen Shot 2021-08-25 at 12 19 32 PM

### CODE REVIEW

**Review by suho — APPROVED** [review]

**Comment by minhnimble** [comment]
Need to fix conflicts in this PR 🛠🦀

**Comment by llleyelll** [comment]
> Need to fix conflicts in this PR 🛠🦀

@minhnimble Conflict resolved, please help check again 🙏

File: fastlane/Fastfile [hunk 1]
```diff
@@ -36,6 +41,25 @@ end
 default_platform(:ios)
 
 platform :ios do
+  # Testing
+
+  desc 'Build for testing'
+  lane :build_for_testing do
```
Comment (suho) [inline fastlane/Fastfile:47]: @llleyelll I think we can use these lanes for CI files also, please help me to update the CI files

**Review by suho — APPROVED** [review]

File: fastlane/Fastfile (hunk 1 above)
Comment (llleyelll) [reply fastlane/Fastfile:47]: @suho Do you mean to update this approach to using those lanes? Am I understand it correctly? 😅
```
- name: Build and Test
  run: xcodebuild -workspace {workspace_name}.xcworkspace -scheme '{staging_scheme_name}' -destination 'platform=iOS Simulator,name=iPhone 12' clean test | xcpretty && exit ${PIPESTATUS[0]}
```

Comment (suho) [reply fastlane/Fastfile:47]: Yes, as we are using `fastlane scan` for testing, so we can update it for the `Build and Test` steps.
I prefer we have some think like this
```rb

lane :build_and_test do
  build_for_testing
  tests
  ui_tests
end
```

then:
```yml
- name: Build and Test
  run: bundle exec fastlane build_and_test
```

Comment (llleyelll) [reply fastlane/Fastfile:47]: Updated in [d5dc2a9](https://github.com/nimblehq/ios-templates/pull/170/commits/d5dc2a9ac1ba7bc3dc2db5d54a375b4e675a0298) 🙏

File: .github/workflows/deploy_AppStore.yml [hunk 2]
```diff
@@ -61,8 +61,8 @@ jobs:
       shell: bash
 
     - name: Build and Test
-      run: xcodebuild -workspace {workspace_name}.xcworkspace -scheme '{staging_scheme_name}' -destination 'platform=iOS Simulator,name=iPhone 12' clean test | xcpretty && exit ${PIPESTATUS[0]}
-
+      run: bundle exec fastlane build_and_test
+
```
Comment (blyscuit) [inline .github/workflows/deploy_AppStore.yml]: please clean up line break indent.

**Review by blyscuit — APPROVED** [review]

File: .github/workflows/deploy_AppStore.yml (hunk 2 above)
Comment (llleyelll) [reply .github/workflows/deploy_AppStore.yml]: Updated in [13eb9c7](https://github.com/nimblehq/ios-templates/pull/170/commits/13eb9c733b22970de2430938b379219616453a99) 🙇‍♀️

**Review by minhnimble — APPROVED** [review]

**Review by suho — APPROVED** [review]

**Pull request merged**